src/dlls/mscoree/unixinterface.cpp

```diff
@@ -146,6 +138,61 @@ int coreclr_initialize(
     ConstWStringArrayHolder propertyValuesW;
     propertyValuesW.Set(StringArrayToUnicode(propertyCount, propertyValues), propertyCount);
 
+    // Process properties that specify CoreCLR startup flags and remove them from the property list
+    STARTUP_FLAGS startupFlags =
+        static_cast<STARTUP_FLAGS>(
+            STARTUP_FLAGS::STARTUP_LOADER_OPTIMIZATION_SINGLE_DOMAIN |
+            STARTUP_FLAGS::STARTUP_SINGLE_APPDOMAIN);
+    int newPropertyIndex = 0;
+    for (int oldPropertyIndex = 0; oldPropertyIndex < propertyCount; ++oldPropertyIndex)
+    {
+        if (wcscmp(propertyKeysW[oldPropertyIndex], W("CONCURRENT_GC")) == 0)
+        {
+            if (wcscmp(propertyValuesW[oldPropertyIndex], W("1")) == 0)
+            {
+                startupFlags = static_cast<STARTUP_FLAGS>(startupFlags | STARTUP_CONCURRENT_GC);
```

Comment:
We should be providing the most appropriate default behavior. Concurrent GC is the most appropriate default behavior. (BTW: It is how desktop works - you get concurrent GC by default.)

This is a problem in the legacy COM hosting interface - that we are nicely fixing here:
- The host has to pick the right values for everything and people get it wrong half of the time, e.g. they do not realize that they have to get all flags right even if they only care about STARTUP_LOADER_OPTIMIZATION.
- The most appropriate defaults do change over time as the runtime evolves. The COM interface does not distinguish between whether the host just wanted a default; or whether it really cares about the flag. (There are creative workarounds around this in desktop.)  


Wrt. testing: We have to enable the concurrent GC by default before RTM. It is fine to stage it if we have signs that it is too destabilizing.